Rewrite Contributing and readme (#4416)
opensearch-project/documentation-website: README.md

# About the OpenSearch documentation repo

> The documentation repository contains the documentation for OpenSearch, the highly scalable and extensible open-source software suite for search, analytics, observability, and other data-intensive applications. You can find the rendered documentation at [opensearch.org/docs](https://opensearch.org/docs).

Reviewer: Should this be the actual name of the repo ("documentation-website")? Do we think we need to include the added text describing OpenSearch, or would the reader likely already know what OpenSearch is if they're reading this file?

## Contributing

> Community contributions remain essential in keeping the documentation comprehensive, useful, well-organized, and up-to-date. If you are interested in submitting an issue or contributing content, see [CONTRIBUTING](CONTRIBUTING.md). 

Reviewer: ```suggestion
Community contributions remain essential to keeping the documentation comprehensive, useful, well organized, and up to date. If you are interested in submitting an issue or contributing content, see [CONTRIBUTING](CONTRIBUTING.md). 
```

> See these important resources when you are considering contributing to the documentation:  

Reviewer: ```suggestion
The following resources provide important guidance regarding contributions to the documentation:  
```

> - [OpenSearch Project Style Guidelines](STYLE_GUIDE.md) - The style guide covers the style standards to be observed when creating OpenSearch content.

Reviewer: ```suggestion
- [OpenSearch Project Style Guidelines](STYLE_GUIDE.md) -- The style guide covers the style standards to be observed when creating OpenSearch Project content.
```

> - [Terms](TERMS.md) - The terms list contains the key OpenSearch terms and tips on how and when to use them.  

Reviewer: ```suggestion
- [OpenSearch terms](TERMS.md) -- The terms list contains key OpenSearch terms and tips on how and when to use them.  
```

> - [API Style Guide](API_STYLE_GUIDE.md) - The API style guide provides the basic structure for creating OpenSearch API documentation.

Reviewer: ```suggestion
- [API Style Guide](API_STYLE_GUIDE.md) -- The API Style Guide provides the basic structure for creating OpenSearch API documentation.
```

> - [Formatting Guide](FORMATTING_GUIDE.md) - The OpenSearch documentation uses a modified version of the [just-the-docs](https://github.com/pmarsceill/just-the-docs) Jekyll theme. Use the formatting guide for an overview of the commonly used formatted elements and to learn how to add pages to the website.

Reviewer: ```suggestion
- [Formatting Guide](FORMATTING_GUIDE.md) -- The OpenSearch documentation uses a modified version of the [just-the-docs](https://github.com/pmarsceill/just-the-docs) Jekyll theme. The Formatting Guide provides an overview of the commonly used formatting elements and how to add a page to the website.
```

The documentation repository contains the documentation for OpenSearch, the search, analytics, and visualization suite with advanced security, alerting, SQL support, automated index management, deep performance analysis, and more. You can find the rendered documentation at [opensearch.org/docs](https://opensearch.org/docs).

## How the website works

> This repository contains [Markdown](https://guides.github.com/features/mastering-markdown/) files organized into Jekyll _collections_ (for example, `_api-reference` or `_dashboards`). Each Markdown file corresponds to one page on the website.

Reviewer 2: Do we need this section here? Would it be more appropriate in the contribution file?

Author: The readme is the what and the contributing is the how in my opinion. So readme is about this repo and contributing is about how to contribute. So I think it's better to have it here.

- [vagimeli](https://github.com/vagimeli)

## Code of conduct

This project has adopted an [Open Source Code of Conduct](https://opensearch.org/codeofconduct.html).

## Security

> If you discover a potential security issue in this project we ask that you notify AWS/Amazon Security using our [vulnerability reporting page](http://aws.amazon.com/security/vulnerability-reporting/). Do **not** create a public GitHub issue.

Reviewer: ```suggestion
If you discover a potential security issue in this project, we ask that you notify AWS/Amazon Security using our [vulnerability reporting page](http://aws.amazon.com/security/vulnerability-reporting/). Do **not** create a public GitHub issue.
```

> If you discover a potential security issue in this project we ask that you notify AWS/Amazon Security using our [vulnerability reporting page](http://aws.amazon.com/security/vulnerability-reporting/). Do **not** create a public GitHub issue.

Reviewer: Re: "AWS/Amazon Security": Which one? Also, if he hasn't already, I would consider having Weston review this statement, especially given that we don't appear to provide a "why" in relation to the last sentence.

Author: This was also taken from the main OpenSearch repo.

Reviewer: Let's just have Weston take a quick look to ensure he's OK with it.

> If you discover a potential security issue in this project we ask that you notify AWS/Amazon Security using our [vulnerability reporting page](http://aws.amazon.com/security/vulnerability-reporting/). Please do **not** create a public GitHub issue.

Reviewer 3: ```suggestion
If you discover a potential security issue in this project we ask that you notify AWS/Amazon Security using our [vulnerability reporting page](http://aws.amazon.com/security/vulnerability-reporting/). Do **not** create a public GitHub issue.
```
